runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/streaming/ActiveWorkStateTest.java

```diff
@@ -0,0 +1,289 @@
+    ActivateWorkResult activateWorkResult =
+        activeWorkState.activateWorkForKey(shardedKey, createWork(createWorkItem(2L)));
+
+    assertEquals(ActivateWorkResult.QUEUED, activateWorkResult);
+  }
+
+  @Test
+  public void testCompleteWorkAndGetNextWorkForKey_noWorkQueueForKey() {
+    assertEquals(
+        Optional.empty(),
+        activeWorkState.completeWorkAndGetNextWorkForKey(shardedKey("someKey", 1L), 10L));
+  }
+
+  @Test
+  public void testCompleteWorkAndGetNextWorkForKey_currentWorkInQueueDoesNotMatchWorkToComplete() {
+    long workTokenToComplete = 1L;
+
+    Work workInQueue = createWork(createWorkItem(2L));
+    ShardedKey shardedKey = shardedKey("someKey", 1L);
+
+    activeWorkState.activateWorkForKey(shardedKey, workInQueue);
+    activeWorkState.completeWorkAndGetNextWorkForKey(shardedKey, workTokenToComplete);
+
+    assertEquals(1, readOnlyActiveWork.get(shardedKey).size());
+    assertEquals(workInQueue, readOnlyActiveWork.get(shardedKey).peek());
+  }
+
+  @Test
+  public void testCompleteWorkAndGetNextWorkForKey_removesWorkFromQueueWhenComplete() {
+    long workTokenToComplete = 1L;
+
+    Work activeWork = createWork(createWorkItem(workTokenToComplete));
+    Work nextWork = createWork(createWorkItem(2L));
+    ShardedKey shardedKey = shardedKey("someKey", 1L);
+
+    activeWorkState.activateWorkForKey(shardedKey, activeWork);
+    activeWorkState.activateWorkForKey(shardedKey, nextWork);
+    activeWorkState.completeWorkAndGetNextWorkForKey(shardedKey, workTokenToComplete);
+
+    assertEquals(nextWork, readOnlyActiveWork.get(shardedKey).peek());
+    assertEquals(1, readOnlyActiveWork.get(shardedKey).size());
+    assertFalse(readOnlyActiveWork.get(shardedKey).contains(activeWork));
+  }
+
+  @Test
+  public void testCompleteWorkAndGetNextWorkForKey_removesQueueIfNoWorkPresent() {
+    Work workInQueue = createWork(createWorkItem(1L));
+    ShardedKey shardedKey = shardedKey("someKey", 1L);
+
+    activeWorkState.activateWorkForKey(shardedKey, workInQueue);
+    activeWorkState.completeWorkAndGetNextWorkForKey(
+        shardedKey, workInQueue.getWorkItem().getWorkToken());
+
+    assertFalse(readOnlyActiveWork.containsKey(shardedKey));
+  }
+
+  @Test
+  public void testCompleteWorkAndGetNextWorkForKey_returnsWorkIfPresent() {
+    Work workToBeCompleted = createWork(createWorkItem(1L));
+    Work nextWork = createWork(createWorkItem(2L));
+    ShardedKey shardedKey = shardedKey("someKey", 1L);
+
+    activeWorkState.activateWorkForKey(shardedKey, workToBeCompleted);
+    activeWorkState.activateWorkForKey(shardedKey, nextWork);
+    activeWorkState.completeWorkAndGetNextWorkForKey(
+        shardedKey, workToBeCompleted.getWorkItem().getWorkToken());
+
+    Optional<Work> nextWorkOpt =
+        activeWorkState.completeWorkAndGetNextWorkForKey(
+            shardedKey, workToBeCompleted.getWorkItem().getWorkToken());
+
+    assertTrue(nextWorkOpt.isPresent());
+    assertSame(nextWork, nextWorkOpt.get());
```

Comment:
would be good to then complete this returned work and verify that works

Reply:
done